### PR TITLE
Update minSDKVersion from API 23 to 24.

### PR DESCRIPTION
**Pull request by brandonpage**
I tried to fix/update everything API 23/24 related that I could find.

### CODE REVIEW

**Comment by brandonpage** [comment]
@bhariharan Is there any work here to be done around around hardware backed keystore?  Android 6 added a lot of security features not present beforehand.

File: .circleci/config.yml [hunk 1]
```diff
@@ -739,46 +738,3 @@ workflows:
           test_apk: *salesforceReact-apk-path
           api_level: 25
 
-  test-api-23:
```
Comment (brandonpage) [inline .circleci/config.yml:742]: Side note:  This file has no business being 700+ lines long.  I will come back to this and fix it up with matrix jobs and/or CircleCI's new monorepo support.

File: libs/SalesforceSDK/build.gradle [hunk 2]
```diff
@@ -11,34 +11,22 @@ dependencies {
     api project(':libs:SalesforceAnalytics')
     api 'com.squareup.okhttp3:okhttp:3.12.12'
     api 'com.google.firebase:firebase-messaging:20.1.0'
```
Comment (brandonpage) [inline libs/SalesforceSDK/build.gradle:13]: The reasons we haven't updated `okhttp` and `firebase-messagine` to the latest are unrelated from API version, correct?

**Comment by bhariharan** [comment]
@brandonpage Are we cleared to bump this up? Have we run external comms and checked with SApp?

**Comment by bhariharan** [comment]
Also, you need to update the tests/test projects too.

File: libs/SalesforceSDK/build.gradle (hunk 2 above)
Comment (bhariharan) [reply libs/SalesforceSDK/build.gradle:13]: Yes, they're unrelated, and we shouldn't bump them up yet.

**Comment by bhariharan** [comment]
I'll take a look at the keystore stuff after this goes in.

**Comment by brandonpage** [comment]
> @brandonpage Are we cleared to bump this up? Have we run external comms and checked with SApp?

I haven't reached out to them but from looking at their repos it seems SApp and SFS are both min 24 already.  I will double check.

**Comment by bhariharan** [comment]
Fair enough. Communities is still on `23`, but I'm okay bumping that up.

File: libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/ui/SalesforceWebViewClient.java [hunk 3]
```diff
@@ -90,21 +90,14 @@ public SalesforceWebViewClient(SalesforceWebViewEngine parentEngine) {
...
-    }
-
     @Override
 	public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
-		return shouldOverrideUrlLoading(view, request.getUrl().toString());
+    	String url = request.getUrl().toString();
+		if (SalesforceWebViewClientHelper.shouldOverrideUrlLoading(ctx, view, url)) {
+			return true;
+		} else {
+			return super.shouldOverrideUrlLoading(view, url);
```
Comment (bhariharan) [inline libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/ui/SalesforceWebViewClient.java:99]:
```suggestion
			return super.shouldOverrideUrlLoading(view, request);
```

**Review by bhariharan — CHANGES_REQUESTED** [review]
Just one change to make.

**Review by bhariharan — APPROVED** [review]

**Pull request merged**